src/services/Game/Game.ts

```diff
@@ -86,171 +107,207 @@ export class Game {
+        );
+
+        let gamepadMovementX = null;
+        let gamepadJump = null;
+
+        const gamepad = navigator.getGamepads()[0];
+
+        if (gamepad) {
+            gamepadMovementX = Math.abs(gamepad?.axes[0]) > GAMEPAD_MOVEMENT_THRESHOLD
+                ? gamepad?.axes[0]
+                : null;
+            gamepadJump = gamepad?.buttons[0].pressed;
+        }
 
         const {
             top, right, bottom, left,
         } = this.canvasSides;
 
         const { x: ballX, y: ballY } = this.ballPosition;
 
-        const closestFloor = this.map.closestFloor(ballX, ballY, bottom - top);
+        const ballCanFall = this.ballMovementChecker.ballCanFall(ballX, ballY);
+
+        const isBallCanGoLeft = this.ballMovementChecker.isBallCanGoLeft(
+            ballX,
+            ballY,
+        );
+
+        const isBallCanGoRight = this.ballMovementChecker.isBallCanGoRight(
+            ballX,
+            ballY,
+        );
+
+        const isNextLvlReached = this.ballMovementChecker.checkNextLvlReached(
+            ballX,
+            ballY,
+        );
+
+        if (isNextLvlReached) {
+            this.moveToNextLvl(isNextLvlReached);
+        }
 
         if (
-            this.velY < MAX_GAMER_SPEED
+            this.velY < MAX_PLAYER_SPEED
             && this.jumpBust === 0
-            && ballY < closestFloor - GAMER_RAD
+            && ballCanFall
         ) {
             this.velY += 2;
         }
 
-        if (this.keys.get('ArrowRight') || this.keys.get('d')) {
-            if (this.velX < MAX_GAMER_SPEED) {
+        if (
+            this.keys.get('ArrowRight')
+            || this.keys.get('d')
+            || (gamepadMovementX && gamepadMovementX > GAMEPAD_MOVEMENT_THRESHOLD)
+        ) {
+            if (this.velX < MAX_PLAYER_SPEED) {
                 this.velX += +LEFT_RIGHT_SPEED_BUST;
             }
         }
 
-        if (this.keys.get('ArrowUp') || this.keys.get('w')) {
+        if (this.keys.get('ArrowUp') || this.keys.get('w') || gamepadJump) {
             if (
-                this.velY > 0
-                && this.velY < 0.0001
-                && this.velY > -MAX_GAMER_SPEED
+                this.ballMovementChecker.isBallCanJump(ballX, ballY)
+                && this.velY >= 0
+                && this.velY > -MAX_PLAYER_SPEED
                 && this.jumpBust === 0
+                && !ballCanFall
             ) {
-                this.velY -= JUMP_VELOCITY;
+                this.velY = -JUMP_VELOCITY;
+
+                // "заряжаем" количество кадров для перемещения вверх
                 this.jumpBust = JUMP_BUST_LIMIT;
             }
         }
 
-        if (this.keys.get('ArrowLeft') || this.keys.get('a')) {
-            if (this.velX > -MAX_GAMER_SPEED) {
+        if (
+            this.keys.get('ArrowLeft')
+            || this.keys.get('a')
+            || (gamepadMovementX && gamepadMovementX < -GAMEPAD_MOVEMENT_THRESHOLD)
+        ) {
+            if (this.velX > -MAX_PLAYER_SPEED) {
```

Comment:
🧐
Унарный минус легко пропустить, так нагляднее. На твой вкус
```suggestion
            || (gamepadMovementX && gamepadMovementX < GAMEPAD_MOVEMENT_THRESHOLD * -1)
        ) {
            if (this.velX > MAX_PLAYER_SPEED * -1) {
```